source/fab/metrics.py

```diff
@@ -242,14 +242,18 @@ def metrics_summary(metrics_folder: Path):
     run = metrics['run']
     time_taken = datetime.timedelta(seconds=int(run['time taken']))
     min_label_thresh = time_taken.seconds * 0.01
-    step_metrics = metrics['steps'].items()
-    step_times = [kv[1] for kv in step_metrics]
-    step_labels = [kv[0] if kv[1] > min_label_thresh else "" for kv in step_metrics]
-
-    plt.pie(step_times, labels=step_labels, normalize=True,
-            wedgeprops={"linewidth": 1, "edgecolor": "white"})
-    plt.suptitle(f"{run['label']} took {time_taken}\n"
-                 f"on {run['sysname']}, {run['nodename']}, {run['machine']}")
-    plt.figtext(0.99, 0.01, f"{metrics['run']['datetime']}", horizontalalignment='right', fontsize='x-small')
-    plt.savefig(metrics_folder / "pie.png")
-    plt.close()
+    step_totals = metrics.get('steps')
```

Comment:
Is this change mentioned in the summary? What's it about?

Reply:
It's a bit of robustness. The new test only has one step, which deliberately fails, so there's never any step timings in the metrics. This change stops that from causing a key error. I'll add this to the description too.

Reply:
You never got round to it but we have a backlog of changes so maybe for a future ticket?